### PR TITLE
Route summary reporting error fixed

### PR DESCRIPTION
**Pull request by rakeshhotker**
As told, i reverted to the Route summary reporting error so as to focus on one bug.

### CODE REVIEW

File: src/Controls/Directions/Summary.jsx [hunk 1]
```diff
@@ -22,38 +21,22 @@ class Summary extends React.Component {
...
-    }
-    if (duration.seconds > 0) {
-      durationStr += duration.seconds + 'sec'
-    }
-    return durationStr
+    const date = new Date(durationInSeconds * 1000)
+    const days = date.getDate() - 1 > 0 ? date.getDate() - 1 + 'd ' : ''
+    const hours = date.getHours() > 0 ? date.getHours() + 'h ' : ''
+    const minutes = date.getMinutes() > 0 ? date.getMinutes() + 'min' : ''
+    return days + hours + minutes
```
Comment (nilsnolde) [inline src/Controls/Directions/Summary.jsx:28]: not DRY, please outsource and re-use

**Review by nilsnolde — CHANGES_REQUESTED** [review]

File: src/Controls/Directions/Summary.jsx (hunk 1 above)
Comment (nilsnolde) [reply src/Controls/Directions/Summary.jsx:28]: although, now there's #91 , so likely the other place where this is done, will change to a hh:mm:ss format. maybe we should do that here too?

Comment (nilsnolde) [reply src/Controls/Directions/Summary.jsx:28]: ended up not using hh:mm:ss format, but compress it a bit regardless in #94.  we should re-use that function.

**Comment by rakeshhotker** [comment]
Could you please be more descriptive

**Comment by nilsnolde** [comment]
You copy/pasted this function as-is from another place in the same codebase. You're repeating the same stuff, that's never good.

Did you look at #94? You'll need to merge master, then you can apply your changes while having the function live somewhere else and called by both modules.

**Comment by aaryahjolia** [comment]
Can I make a PR on this @nilsnolde ? I solved this issue by DRY principle and in hh:mm:ss format.

**Comment by nilsnolde** [comment]
Depends if @rakeshhotker is dropping the contribution or not, I'd still give him some time.

**Pull request closed without merging**